File: clients/algoliasearch-client-php/lib/Model/Recommend/AdvancedSyntaxFeatures.php
<?php

namespace Algolia\AlgoliaSearch\Model\Recommend;

/**
 * AdvancedSyntaxFeatures Class Doc Comment
 *
 * @category Class
 * @package Algolia\AlgoliaSearch
 */
class AdvancedSyntaxFeatures
{
    /**
     * Possible values of this enum
     */
    const EXACT_PHRASE = 'exactPhrase';

    const EXCLUDE_WORDS = 'excludeWords';

    /**
     * Gets allowable values of the enum
     *
     * @return string[]
     */
    public static function getAllowableEnumValues()
    {
        return [
            self::EXACT_PHRASE,
            self::EXCLUDE_WORDS,
        ];
    }
}

File: clients/algoliasearch-client-java-2/algoliasearch-core/src/com/algolia/model/search/AdvancedSyntaxFeatures.java
package com.algolia.model.search;

import com.google.gson.TypeAdapter;
import com.google.gson.annotations.JsonAdapter;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;
import java.io.IOException;

/** Gets or Sets advancedSyntaxFeatures */
@JsonAdapter(AdvancedSyntaxFeatures.Adapter.class)
public enum AdvancedSyntaxFeatures {
  EXACT_PHRASE("exactPhrase"),

  EXCLUDE_WORDS("excludeWords");

  private final String value;

  AdvancedSyntaxFeatures(String value) {
    this.value = value;
  }

  public String getValue() {
    return value;
  }

  @Override
  public String toString() {
    return String.valueOf(value);
  }

  public static AdvancedSyntaxFeatures fromValue(String value) {
    for (AdvancedSyntaxFeatures b : AdvancedSyntaxFeatures.values()) {
      if (b.value.equals(value)) {
        return b;
      }
    }
    throw new IllegalArgumentException("Unexpected value '" + value + "'");
  }

  public static class Adapter extends TypeAdapter<AdvancedSyntaxFeatures> {

    @Override
    public void write(
      final JsonWriter jsonWriter,
      final AdvancedSyntaxFeatures enumeration
    ) throws IOException {
      jsonWriter.value(enumeration.getValue());
    }

    @Override
    public AdvancedSyntaxFeatures read(final JsonReader jsonReader)
      throws IOException {
      String value = jsonReader.nextString();
      return AdvancedSyntaxFeatures.fromValue(value);
    }
  }
}
